rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -688,8 +697,11 @@ protected MavenRepository normalizeRepository(MavenRepository originalRepository
                                 } else if (!e.isClientSideException()) {
                                     return originalRepository;
                                 }
-                            } catch (Throwable ignored) {
+                            } catch (Throwable e) {
                                 // ok to fall through here and cache a null
+                                if (nullReasonConsumer != null) {
+                                    nullReasonConsumer.accept(t.getMessage());
+                                }
```

Comment:
Is this really a "null reason consumer" if it also handles other cases?
In this case would you prefer to get access to the whole exception, not just the message?

Maybe you want `Consumer<Exception> onError` ? Can always wrap a string in a `RuntimeException` in the actual `null` case